src/transformers/quantizers/auto.py

```diff
@@ -195,10 +199,13 @@ def merge_quantization_configs(
             warning_msg = ""
 
         if isinstance(quantization_config, dict):
-            quantization_config = AutoQuantizationConfig.from_dict(quantization_config)
+            if quantization_config_from_args.__class__.__name__ == "AutoRoundConfig":
+                quantization_config = AutoRoundConfig.from_dict(quantization_config)
+            else:
+                quantization_config = AutoQuantizationConfig.from_dict(quantization_config)
```

Comment:
auto quantization config should successfully dispatch to it, no need for this if/else

Reply:
We support forcibly converting models in the AutoAWQ and GPTQ formats to the AutoRound format, enabling improved inference support on CPU and XPU. For instance, this allows us to run INT2 inference on CPU. An example usage is: 
~~~python
  from transformers import AutoRoundConfig
  quantization_config = AutoRoundConfig()
  model_name = "ybelkada/opt-125m-gptq-4bit"
  model = AutoModelForCausalLM.from_pretrained(model_name,
                                               device_map="auto",quantization_config=quantization_config)
  tokenizer = AutoTokenizer.from_pretrained(model_name)
~~~

Reply:
maybe use isintance and comment how `quantization_config_from_args` takes precedence and can convert `quantization_config` in the case of `AutoRoundConfig`

Reply:
Also notice that you are supposing that `quantization_config_from_args` is not None, the if statement will fail if it is.

Reply:
Thank you for the reminder. I've switched to using `isinstance` for type checking.